src/update.c

```diff
@@ -147,27 +147,39 @@ update_state_free (updateStatePtr updateState)
 	g_free (updateState);
 }
 
-/* update request processing */
+/* update request object */
 
-updateRequestPtr
-update_request_new (void)
-{
-	return g_new0 (struct updateRequest, 1);
-}
+G_DEFINE_TYPE (UpdateRequest, update_request, G_TYPE_OBJECT);
 
-void
-update_request_free (updateRequestPtr request)
+static void
+update_request_finalize (GObject *obj)
 {
-	if (!request)
-		return;
+	UpdateRequest *request = UPDATE_REQUEST (obj);
 
 	update_state_free (request->updateState);
 	update_options_free (request->options);
 
 	g_free (request->postdata);
 	g_free (request->source);
 	g_free (request->filtercmd);
-	g_free (request);
+}
```

Comment:
I think the parent class's finalized should be called here. Although I'm not sure if GObject's finalize does anything ..

Reply:
Added it just to be sure!